release/pkg/bundles/package-controller.go

```diff
@@ -191,3 +204,60 @@ func replaceTag(uri, tag string) string {
 	uri = strings.Join(NewURIList[:], ":")
 	return uri
 }
+
+// getTag is used to replace get the tag of an ImageURI
+func getTag(uri string) string {
+	NewURIList := strings.Split(uri, ":")
+	if len(NewURIList) < 2 {
+		return uri
+	}
+	return NewURIList[len(NewURIList)-1]
+}
+
+// recoverBundle is a function used if we have a helm error to revert the bundle back to it's original state using Git Tag version.
+func recoverBundle(r *releasetypes.ReleaseConfig, imageDigests map[string]string, helmtag, imagetag, tokentag string) (map[string]anywherev1alpha1.Image, []string) {
```

Comment:
I'm probably misunderstanding the problem as a whole, but is there a reason we don't just always point the release branch at the prod packages bundle? As a prod build, I feel like it might make sense to explicitly point it at the prod packages bundle?

If we are committed to this recovery logic, could we get a unit test on it?

Reply:
I'm a little unclear on why this is needed too. If packages is meant to be an independently maintained and releasable artifact, would we not want to depend on a stable release in EKS-A?

Reply:
So this is a scenario where the dev-packages pipeline runs into issues and is failing, we don't want to block the EKS-A team or e2e tests since the package controller install is part of the cluster creation process.